problem4.js

```diff
@@ -0,0 +1,212 @@
+];
+let HISTORY_income = [
+  { category: "용돈", detail: "추석용돈", money: 150000 },
+];
+
+const CATEGORIES = {
+  '수입': ['용돈', '알바비', '환불액'],
+  '지출': ['여행비', '식사비', '교통비']
+};
+
+const emptylist = [];
+function Showlist(selection) {
+  INIT_BALANCE = 0;
+  const balanceElement = document.querySelector(".totalsum strong");
+  const detailsSection = document.querySelector(".details");
+  let AllInputadd = document.querySelector(".incomevalue");
+  let AllOutputadd = document.querySelector(".outcomevalue");
+  let AllInput = 0 ;
+  let AllOutput = 0 ;
+  detailsSection.innerHTML = "";
+
+  for (const historyItem of selection) {
+//------------- 초기값 업데이드
+    INIT_BALANCE += historyItem.money;
+
+    const itemElement = document.createElement("div");
+    itemElement.classList.add("item");
+
+    const categoryElement = document.createElement("span");
+    categoryElement.classList.add("category");  
+    categoryElement.textContent = historyItem.category;
+
+    const detailElement = document.createElement("strong");
+    detailElement.classList.add("detail");
+    detailElement.textContent = historyItem.detail;
+
+    const moneyElement = document.createElement("span");
+    moneyElement.classList.add(
+      "money_" + (historyItem.money >= 0 ? "plus" : "minus")
+    );
+    moneyElement.textContent = " " + Math.abs(historyItem.money);
+
+    const deleteButton = document.createElement("button");
+    deleteButton.classList.add("x");
+    deleteButton.textContent = "x";
+          
+    if(historyItem.money>0){
+      AllInput += historyItem.money;
+    }
+    else{
+      AllOutput += Math.abs(historyItem.money);
+    }
+//                        click관련된 함수 
+    deleteButton.onclick=function(){
+      detailsSection.removeChild(itemElement);
+      INIT_BALANCE -= historyItem.money;
+      balanceElement.textContent = INIT_BALANCE;
+      
+      if(historyItem.money>0){
+        AllInput -= historyItem.money;
+        AllInputadd.textContent ="총 수입 "+AllInput;
+      }
+      else{
+        AllOutput -= Math.abs(historyItem.money);
+        AllOutputadd.textContent ="총 지출 "+AllOutput;
+      }
+      
+    };
+
+    itemElement.appendChild(categoryElement);
+    itemElement.appendChild(detailElement);
+    itemElement.appendChild(moneyElement);
+    itemElement.appendChild(deleteButton);
+    detailsSection.appendChild(itemElement);
+  }
+  balanceElement.textContent = INIT_BALANCE;
+  AllInputadd.textContent ="총 수입 "+AllInput;
+  AllOutputadd.textContent ="총 지출 "+AllOutput;
+
+
+}
+Showlist(HISTORY_LIST);
+
+const inputBox = document.getElementById("input_box");
+const outputBox = document.getElementById("output_box");
+
+function filtering() {
+  if (inputBox.checked && outputBox.checked) {
+    Showlist(HISTORY_LIST);
+  } else if (inputBox.checked) {
+    Showlist(HISTORY_income);
+  } else if (outputBox.checked) {
+    Showlist(HISTORY_spend);
+  } else {
+    Showlist(emptylist);
+  }
+}
```

Comment:
마쟈마쟈 혜인언니 말처럼 switch를 쓰거나 if를 쓰려면 중첩을 더 최소화 할 수 있을 것 같아
```suggestion
function filtering() {
  if (inputBox.checked) {
    if (outputBox.checked) {
      Showlist(HISTORY_LIST);
    } else {
      Showlist(HISTORY_income);
    }
  } else if (outputBox.checked) {
    Showlist(HISTORY_spend);
  } else {
    Showlist(emptylist);
  }
}
```